libcontainer/process_linux.go

```diff
@@ -571,7 +571,7 @@ func (p *initProcess) start() (retErr error) {
 					return fmt.Errorf("error setting Intel RDT config for procHooks process: %w", err)
 				}
 			}
-			if p.config.Config.Hooks != nil {
+			if p.config.Config.Hooks != nil && len(p.config.Config.Hooks) != 0 {
```

Comment:
Since `len(nil)` returns `0`, you can simplify the check (removing the first one).

LGTM otherwise

Reply:
Maybe `len() > 0` is slightly better idiomatically, although either way is fine for me.

Reply:
yes, that's seems better.